crates/bevy_core_pipeline/src/bloom/mod.rs

```diff
@@ -32,15 +39,57 @@ use std::num::NonZeroU32;
 const BLOOM_SHADER_HANDLE: HandleUntyped =
     HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 929599476923908);
 
+const BLOOM_TEXTURE_FORMAT: TextureFormat = TextureFormat::Rg11b10Float;
+
+impl BloomSettings {
+    /// Calculates blend intensities of blur pyramid levels
+    /// during the upsampling+compositing stage.
+    ///
+    /// The function assumes all pyramid levels are upsampled and
+    /// blended into higher frequency ones using this function to
+    /// calculate blend levels every time. The final (highest frequency)
+    /// pyramid level in not blended into anything therefore this function
+    /// is not applied to it. As a result, the *mip* Earameter of 0 indicates
+    /// the second-highest frequency pyramid level (in our case that is the
+    /// 0th mip of the bloom texture with the original image being the
+    /// actual highest frequency level).
+    ///
+    /// Parameters:
+    /// * *mip* - the index of the lower frequency pyramid level (0 - max_mip, where 0 indicates highest frequency mip but not the highest frequency image).
+    /// * *max_mip* - the index of the lowest frequency pyramid level.
+    ///
+    /// This function can be visually previewed for all values of *mip* (normalized) with tweakable
+    /// BloomSettings parameters on [Desmos graphing calculator](https://www.desmos.com/calculator/ncc8xbhzzl).
+    fn compute_blend_factor(&self, mip: f32, max_mip: f32) -> f32 {
+        let x = mip / max_mip;
+
+        let mut lf_boost =
+            (1.0 - (1.0 - x).powf(1.0 / (1.0 - self.lf_boost_curvature))) * self.lf_boost;
+        let high_pass_lq =
+            1.0 - ((x - self.high_pass_frequency) / self.high_pass_frequency).clamp(0.0, 1.0);
+        // let high_pass_hq = 0.5 + 0.5 * ((x - self.high_pass_frequency) / self.high_pass_frequency).clamp(0.0, 1.0).mul(std::f32::consts::PI).cos();
+        let high_pass = high_pass_lq;
+
+        lf_boost *= match self.composite_mode {
```

Comment:
Replace this multiply match with an if statement checking for the first case, and then multiplying if true.

Reply:
Oh no, this must look so dumb out of context 😆

I did it that way because this function should eventually end up in a shader (when lens dirt or UI blur is needed) so i tried to avoid a branching statement in favor of a something closer to a ternary operator. Do you still think i should change it?

Reply:
You: "Please add a comment explaining as such then :)"

Reply:
Yeah, still change it for now. If it ends up in the shader, we can change it then.